core/tests/protractor_utils/BlogDashboardPage.js

```diff
@@ -0,0 +1,368 @@
+
+  this.getMatTab = async function(tabName) {
+    for (i = 0; i < await matTabLabels.count(); i++) {
+      var matTab = matTabLabels.get(i);
+      var tabText = await action.getText(
+        `Blog Dashboard tab ${i}`, matTab);
+      if (tabText.startsWith(tabName)) {
+        await action.click(`${tabName} tab`, matTab);
+      }
+    }
+  };
+
+  this.navigateToPublishTab = async function() {
+    await waitFor.pageToFullyLoad();
+    await this.getMatTab('PUBLISHED');
+  };
+
+  this.navigateToDraftsTab = async function() {
+    await waitFor.pageToFullyLoad();
+    await this.getMatTab('DRAFTS');
+  };
+
+  this.expectNumberOfDraftBlogPostsToBe = async function(number) {
+    await this.waitForDraftsBlogPostsToLoad();
+    expect(await draftBlogPostTiles.count()).toBe(number);
+  };
+
+  this.blogDashboardIntroMessageIsVisible = async function() {
+    await waitFor.visibilityOf(
+      blogDashboardIntroMessageContainer, 'Blog Dashboard Intro message ' +
+      'taking too long to be visible');
+  };
+
+  this.expectNumberOfPublishedBlogPostsToBe = async function(number) {
+    await this.waitForPublishedBlogPostsToLoad();
+    expect(await publishedBlogPostTiles.count()).toBe(number);
+  };
+
+  this.expectNumberOfBlogPostsRowsToBe = async function(number) {
+    await this.waitForDraftsBlogPostsToLoad();
+    expect(await blogPostListItems.count()).toBe(number);
+  };
+
+  this.getBlogPostTileEditOption = async function(title) {
+    for (i = 0; i < await blogPostTiles.count(); i++) {
+      var blogPostTile = blogPostTiles.get(i);
+      var blogPostTitleContainer = blogPostTile.element(
+        blogPostTileTitle);
+      var blogPostTitle = await action.getText(
+        `Blog Post Tile Title ${i}`, blogPostTitleContainer);
+      if (blogPostTitle === title) {
+        var blogPostEditOptionButton = blogPostTile.element(
+          blogPostEditOptions);
+        return blogPostEditOptionButton;
+      }
+    }
+  };
+
+  this.deleteBlogPostWithTitle = async function(title) {
+    var blogPostEditOptionButton = await this.getBlogPostTileEditOption(title);
+    await action.click(
+      'Blog post edit option', blogPostEditOptionButton);
+    await action.click('Delete blog post button', deleteBlogPostButton);
+    await action.click(
+      'Confirm Delete Blog Post button', confirmButton);
+    await waitFor.visibilityOfSuccessToast('Blog Post Deleted Successfully.');
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.navigateToBlogPostEditorWithTitleFromList = async function(title) {
+    for (i = 0; i < await blogPostListItems.count(); i++) {
+      var blogPostRow = blogPostListItems.get(i);
+      var blogPostTitleContainer = blogPostRow.element(
+        blogPostTileTitle);
+      var blogPostTitle = await action.getText(
+        `Blog Post Tile Title ${i}`, blogPostTitleContainer);
+      if (blogPostTitle === title) {
+        await action.click('blog post row', blogPostRow);
+        await waitFor.pageToFullyLoad();
```

Comment:
Once we find a match, we don't want to continue the loop, do we?

Reply:
we are returning the function with the value `blogPostEditOptionButton` as soon as the match is found. So it will not continue running, right?

Reply:
I don't see any return statement in this function (`navigateToBlogPostEditorWithTitleFromList`)

Reply:
Sorry, I got confused with the other function. Fixed. Added a `break` statement after 284.